### PR TITLE
feat(builder): create one method per relation type in the collection customizer

### PR DESCRIPTION
**Pull request by romain-gilliotte**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)
- [ ] Ensure that Types have been updated according to your changes (if needed)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/_example/src/customizations/store.ts [hunk 1]
```diff
@@ -2,24 +2,9 @@ import { Collection } from '@forestadmin/agent';
...
-    })
-    .addRelation('dvd', {
-      type: 'OneToMany',
-      foreignCollection: 'dvd',
-      originKey: 'storeId',
-      originKeyTarget: 'id',
-    })
+    .addManyToOne('owner', 'owner', { foreignKey: 'ownerId' })
+    .addOneToOne('address', 'address', { originKey: 'storeId' })
+    .addOneToMany('dvd', 'dvd', { originKey: 'storeId' })
```
Comment (arnaud-moncel) [inline packages/_example/src/customizations/store.ts:7]:
```suggestion
    .addOneToMany('dvds', 'dvd', { originKey: 'storeId' })
```
?

File: packages/agent/test/builder/collection.test.ts [hunk 2]
```diff
@@ -129,24 +129,94 @@ describe('Builder > Collection', () => {
...
-    it('should add a relation', async () => {
+  describe('relations', () => {
+    it('should add a many to one', async () => {
       const { stack, collectionBuilder, collectionName } = await setup();
 
       const collection = stack.relation.getCollection(collectionName);
       const spy = jest.spyOn(collection, 'addRelation');
+      const self = collectionBuilder.addManyToOne('myself', collectionName, {
+        foreignKey: 'firstName',
+        foreignKeyTarget: 'firstName',
```
Comment (arnaud-moncel) [inline packages/agent/test/builder/collection.test.ts:140]: For all tests, I think you need differentiate the options values.
If the code change and for exemple use `options.foreignKey` for the two option `foreignKey` and `foreignKeyTarget` the test should not fail.

File: packages/agent/test/builder/collection.test.ts [hunk 3]
```diff
@@ -129,24 +129,94 @@ describe('Builder > Collection', () => {
...
+        originKey: 'firstName',
+        originKeyTarget: 'firstName',
+      });
 
       expect(spy).toBeCalledTimes(1);
-      expect(spy).toHaveBeenCalledWith('myself', relationDefinition);
+      expect(spy).toHaveBeenCalledWith('myself', {
+        type: 'ManyToMany',
+        foreignCollection: collectionName,
+        throughCollection: collectionName,
```
Comment (arnaud-moncel) [inline packages/agent/test/builder/collection.test.ts:214]: Same here :)

**Review by arnaud-moncel — COMMENTED** [review]
thank's to split it, let move 💪

File: packages/_example/src/customizations/store.ts (hunk 1 above)
Comment (romain-gilliotte) [reply packages/_example/src/customizations/store.ts:7]: 👍

**Review by arnaud-moncel — APPROVED** [review]
🚀

**Pull request merged**